mcs/class/referencesource/mscorlib/system/runtime/serialization/formatterservices.cs

```diff
@@ -146,7 +146,11 @@ static FormatterServices()
             /*out*/ parentTypeCount = 0;
             bool unique = true;
             RuntimeType objectType = (RuntimeType)typeof(object);
-            for (RuntimeType t1 = parentType; t1 != objectType; t1 = (RuntimeType)t1.BaseType)
+            if (parentType == null)
```

Comment:
This is redundant check, https://github.com/mono/mono/blob/ae45caafd9fdbc40fc97d103ca0b1cb046a2e73a/mcs/class/referencesource/mscorlib/system/runtime/serialization/formatterservices.cs#L108 already checks for null